modules/hostap/src/supp_api.c

```diff
@@ -358,13 +358,27 @@ static int wpas_add_and_config_network(struct wpa_supplicant *wpa_s,
 		} else if (params->security == WIFI_SECURITY_TYPE_PSK ||
 				   params->security == WIFI_SECURITY_TYPE_WPA_PSK) {
 			_wpa_cli_cmd_v("set_network %d psk \"%s\"",
-			resp.network_id, params->psk);
+					resp.network_id, params->psk);
 			_wpa_cli_cmd_v("set_network %d key_mgmt WPA-PSK",
-				resp.network_id);
+					resp.network_id);
 			if (params->security == WIFI_SECURITY_TYPE_WPA_PSK) {
 				_wpa_cli_cmd_v("set_network %d proto WPA",
-					resp.network_id);
+						resp.network_id);
+			}
+		} else if (params->security == WIFI_SECURITY_TYPE_WPA_AUTO) {
+			if (params->sae_password) {
+				_wpa_cli_cmd_v("set_network %d sae_password \"%s\"",
+						resp.network_id, params->sae_password);
+			} else {
+				_wpa_cli_cmd_v("set_network %d sae_password \"%s\"",
+						resp.network_id, params->psk);
 			}
```

Comment:
No need to forcibly put `sae_password`, SAE will work with PSK too.

Reply:
Done.